packages/Angular/explorer-core/src/lib/generic-browser-list/generic-browser-list.component.ts

```diff
@@ -0,0 +1,181 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { Router } from '@angular/router'
+import { SharedService } from '@memberjunction/ng-shared';
+import { Folder, Item, ItemType, UpdateItemEvent, UpdateItemEventType } from '../../generic/Item.types';
+import { DashboardCategoryEntity, DashboardEntity } from '@memberjunction/core-entities';
+import { BaseEntity, Metadata, RunView } from '@memberjunction/core';
+
+@Component({
+  selector: 'app-generic-browser-list',
+  templateUrl: './generic-browser-list.component.html',
+  styleUrls: ['./generic-browser-list.component.css', '../../shared/first-tab-styles.css']
+})
+export class GenericBrowserListComponent {
+  @Input() public showLoader: boolean = true;
+  @Input() public itemType: string = '';
+  @Input() public title: string = '';
+  @Input() public items: any[] = [];
+  @Input() public iconName: string = 'view';
+  @Input() public showAddButton: boolean = false;
+  @Input() public addText: string = 'Create New';
+  @Input() public backText: string = 'Go Back';
+  @Input() public EntityName: string = '';
+  @Input() public CategoryName: string = '';
```

Comment:
@JS-BC  is this CategoryName the name of the entity for the category? If so, let's change it to be more descriptive.

Also, please add JSDoc for all the input and output properties for this class as we'll use it a lot in the framework as will others

Reply:
Will do